### DISCUSSION TITLE
Nuxt3 Typecheck is not working on Template

### DISCUSSION THREAD

**Discussion by seyyedhassanpour** [nuxt/nuxt#23301] [Questions] [answered]
I have two components in the `/components/Filter/` directory: `Filter.vue`, `FilterItem.vue`. 
When I import `FilterItem.vue` explicitly in the `Filter.vue`, typechecking in template will work correctly and says you have passed array of number to array of string which is not acceptable. but if I omit `import FilterItem from './FilterItem.vue'` this import from `Filter.vue`, the project still works fine, but the typecheck is not working anymore.

```vue
//Filter.vue
<script setup lang="ts">
import FilterItem from './FilterItem.vue'
</script>

<template>
  <FilterItem :label="[1000]"/>
</template>
```

```vue
// FilterItem.vue
<script setup lang="ts">
import { PropType } from 'vue'

defineProps({
  label: {
    type: Array as PropType<string[]>,
    required: true,
  },
})
</script>

<template>
  <div>
    <h4>{{ label }}</h4>
  </div>
</template>
```

What is the reason and how to fix this issue?
Should we always explicitly import components when we are in the `/component` directory?

**Comment by xjccc** [comment]
when you create some files, are you restart the project ? 

`
pnpm run dev --host
`

my projects is well

![image](https://github.com/nuxt/nuxt/assets/16478776/352c1d32-06f9-4e4b-9174-c8e3b51c4aa0)

**Reply by seyyedhassanpour** [reply]
I'm facing a super weird issue, as you can see in the below picture, I've made 2 folder with diffrent names but with exatcly  the same components, eslint shows error for one but not the other one, could you please test such a case and tell me what was the result?
<img width="1430" alt="image" src="https://github.com/nuxt/nuxt/assets/144664926/3b58c5ea-fce7-438f-afe2-fba2f7aadae9">
 
filter.vue
```vue
<template>
  <MyFilterItem :model-value="[12]" />
  <FilterItem :model-value="[12]" />
</template>
```

FilterItem.vue
```vue
<script setup lang="ts">
import { PropType } from "vue";

defineProps({
  modelValue: {
    type: Array as PropType<string[]>,
    required: true,
  },
});
</script>

<template>
  <div>{{ modelValue }}</div>
</template>

```

**Reply by xjccc** [reply]
emmm...

try to restart your vscode?

- Node Version:     v16.18.0
- Nuxt Version:     3.7.1 or 3.7.3
- CLI Version:      3.8.4
- Nitro Version:    2.6.3
- Package Manager:  pnpm@8.7.0



![image](https://github.com/nuxt/nuxt/assets/16478776/fe092744-8a6b-488b-8f19-9e4ab32174c0)

**Reply by seyyedhassanpour** [reply]
Could you please tell me what extension do you use?
I'm new, maybe the extension I'm using is not the right one

**Reply by xjccc** [reply]
vscode - volar

eslint
```
"eslint": "^8.49.0",
"@typescript-eslint/eslint-plugin": "^6.7.2",
"@typescript-eslint/parser": "^6.7.2",
"eslint-plugin-nuxt": "^4.0.0",
"eslint-plugin-vue": "^9.17.0",
```

**Reply by seyyedhassanpour** [reply]
This is excatly the same I have. 
I've tried to delete `.nuxt` folder and run `yarn dev` again, and now it seems working perfectly. I'm not sure why

**Reply by xjccc** [reply]
> This is excatly the same I have. 
> 
> I've tried to delete `.nuxt` folder and run `yarn dev` again, and now it seems working perfectly. I'm not sure why 

it will regenerate components types

**Comment by seyyedhassanpour** [comment, marked as answer]
This solved me my issue:
I've tried to delete `.nuxt` folder and run yarn dev again, and now it seems working perfectly.
